lib/ansible/module_utils/network/iosxr/utils/utils.py

```diff
@@ -96,6 +96,10 @@ def filter_dict_having_none_value(want, have):
                     if each.get('secondary') and diff_ip is True:
                         test_key_dict.update({'secondary': True})
                     test_dict.update({'ipv4': test_key_dict})
+                # Checks if want doesn't have secondary IP but have has secondary IP set
```

Comment:
so the issue was related with this? I't be great to have some description about the fix around. Thanks!

Reply:
@danielmellado the issue wasn't around this code fix, the problem was with how iosxr handles virtual and logical interfaces, previously I was using virtual and logical interfaces combo in integration tests and now have moved the tests to use zuul physical interface only instead of virtual ones and this resulted in solving the intermittent failures. I came across for the above code fix when I was trying with certain use cases myself, and that's why the respective comment for the fix

Reply:
@justjais great to know! would you mind updating the PR message to include a summary of this? I think it'd be easier afterwards to check what the root case/root fix was.

Besides that the code LGTM!

Reply:
@danielmellado sure, will open an issue with the description and link to the PR. :)